### PR TITLE
Zeppelin jvm mem options specific to intp

### PR DESCRIPTION
**Pull request by piyush-mukati**

### CODE REVIEW

**Comment by piyush-mukati** [comment]
This  change will enable flexibility to configure the memory for each  interpreter.

File: conf/zeppelin-env.sh.template [hunk 1]
```diff
@@ -20,7 +20,9 @@
 # export MASTER=                 		# Spark master url. eg. spark://master_addr:7077. Leave empty if you want to use local mode.
 # export ZEPPELIN_JAVA_OPTS      		# Additional jvm options. for example, export ZEPPELIN_JAVA_OPTS="-Dspark.executor.memory=8g -Dspark.cores.max=16"
 # export ZEPPELIN_MEM            		# Zeppelin jvm mem options Default -Xmx1024m -XX:MaxPermSize=512m
-# export ZEPPELIN_INTP_MEM       		# zeppelin interpreter process jvm mem options. Default = ZEPPELIN_MEM
+# export ZEPPELIN_INTP_DEFAULT_MEM      # zeppelin interpreter process jvm mem options. Default = ZEPPELIN_MEM
+# export {INTERPRETER_ID}_INTP_MEM      # Zeppelin jvm mem options specific to intp
```
Comment (felixcheung) [inline conf/zeppelin-env.sh.template]: could you give an example of {INTERPRETER_ID} here?

Comment (jongyoul) [inline conf/zeppelin-env.sh.template]: Could you please fix the indentation of the comments? And also please remove the empty line below.

File: bin/interpreter.sh [hunk 2]
```diff
@@ -47,6 +47,24 @@ fi
...
 . "${bin}/common.sh"
 
+INTERPRETER_ID=$(basename "${INTERPRETER_DIR}")
+
+
+if [[ -z "${ZEPPELIN_INTP_MEM}" ]]; then
+  export ZEPPELIN_INTP_MEM="${ZEPPELIN_INTP_DEFAULT_MEM}"
+fi
+
+INTERPRETER_SPECIFIC_MEM_VARIABLE=`echo $INTERPRETER_ID | awk '{print toupper($0)"_INTP_MEM" }'`
```
Comment (jongyoul) [inline bin/interpreter.sh]: I think this makes more environments variables like FLINK_INTP_MEM, SPARK_INTP_MEM and so on. I don't think it's good way. It looks good to use the interpreter settings from web UI. What do you think of it?

Comment (Leemoonsoo) [inline bin/interpreter.sh]: I agree. however, i think until interpreter setting page get's such feature, this PR's approach can be useful. And once interpreter setting page have mem configure feature, this env variable can provides default value.

**Comment by Leemoonsoo** [comment]
@piyush-mukati Thanks for great contribution.
It's always good idea to create a JIRA issue for it.  One good reason to have JIRA issue is, Zeppelin generates and publish it's release note from JIRA. So the PR without JIRA issue might not included in the release note.

**Comment by piyush-mukati** [comment]
Hi, 
Sorry for the Delay. i have created the JIRA https://issues.apache.org/jira/browse/ZEPPELIN-393.

**Comment by minahlee** [comment]
@piyush-mukati any update on this PR? I found one bug related to spark interpreter jvm memory setting, and this feature will be useful to handle this bug. Do you mind to reopen this?

**Pull request closed without merging**